File: app/src/main/java/com/saint/struct/ipc/binder/ServerService.java
package com.saint.struct.ipc.binder;

import android.app.Person;
import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.Nullable;

public class ServerService extends Service {
    private static final String TAG = "ServerService";
    private List<Person> mPeople = new ArrayList<>();

    @Override
    public void onCreate() {
        super.onCreate();
        mPeople.add(new Person.Builder().build());
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    private BinderObj mStub = new BinderObj() {
        @Override
        public void addPerson(Person p) {
            if(p == null){
                p = new Person.Builder().build();
            }
            mPeople.add(p);
        }


        @Override
        public List<Person> getPersonList() {
            return mPeople;
        }
    };
}

File: app/src/main/java/com/saint/struct/ipc/binder/ServerService.kt
package com.saint.struct.ipc.binder

import android.app.Service
import android.content.Intent
import android.os.IBinder
import com.saint.struct.ipc.aidl.Person

class ServerService : Service() {
    private val mPeople: MutableList<Person> = ArrayList()
    override fun onCreate() {
        super.onCreate()
        mPeople.add(Person())
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    private val mStub: BinderObj = object : BinderObj() {
        override fun addPerson(p: Person) {
            var p: Person? = p
            if (p == null) {
                p = Person()
            }
            mPeople.add(p)
        }

        override fun getPersonList(): List<Person> {
            return mPeople
        }
    }

    companion object {
        private const val TAG = "ServerService"
    }
}
